crates/driver/src/infra/observe/metrics.rs

```diff
@@ -19,6 +19,10 @@ pub struct Metrics {
     /// The results of the mempool submission.
     #[metric(labels("mempool", "result"))]
     pub mempool_submission: prometheus::IntCounterVec,
+    /// The number of blocks passed between the first time submission was
+    /// atempted and the error detection.
+    #[metric(labels("mempool", "error"))]
+    pub mempool_submission_error_blocks_passed: prometheus::HistogramVec,
```

Comment:
I'm not sure a histogram is useful here. You would definitely need custom buckets (values can only be 0, 1 and 2 unless I'm mistaken). I think having a counter and alerting based on rate is simpler.